### PR TITLE
Remove <Textarea> from <InputComponent>

### PR DESCRIPTION
**Pull request by hotzenklotz**
ATM, our custom `<InputComponent>` supports rendering antd's `<Input>`and `<TextArea>` which leads to very questionable TS types. It turns out that `<TextArea>` is really only used in location, so I extracted it from `<InputComponent>`.

Comments in `<InputComponent>` that React 16 might have fixed the jumping cursors bug and that no custom wrapper for `<Input>` is necessary. I tried that and still had jumping cursors :-/ So, I guess the `<InputComponent>` will be around for a little longer but with simpler typing.

I also updated/fixed some generic types here and there to more specific events.

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Perhaps try the annotation description Markdown modal.
- Run typechecker 

### Issues:
- fixes #

------
(Please delete unneeded items, merge only when none are left open)
- [ ] Updated [(unreleased) changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [ ] Updated [(unreleased) migration guide](../blob/master/MIGRATIONS.unreleased.md#unreleased) if applicable
- [ ] Updated [documentation](../blob/master/docs) if applicable
- [ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes
- [ ] Adapted [wk-libs python client](https://github.com/scalableminds/webknossos-libs/tree/master/webknossos/webknossos/client) if relevant API parts change
- [ ] Needs datastore update after deployment
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/dashboard/advanced_dataset/dataset_table.tsx [hunk 1]
```diff
@@ -657,7 +657,7 @@ export function DatasetTags({
     updatedDataset: APIMaybeUnimportedDataset,
     shouldAddTag: boolean,
     tag: string,
-    event: React.MouseEvent,
+    event: React.SyntheticEvent,
```
Comment (hotzenklotz) [inline frontend/javascripts/dashboard/advanced_dataset/dataset_table.tsx:660]: This callback is used in multiple props. Some are `MouseEvent`s,  some are `ChangeEvent`s, ...

File: frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx [hunk 2]
```diff
@@ -1060,9 +1060,6 @@ class DatasetSettings extends React.PureComponent<DatasetSettingsProps, State> {
             <Row justify="center" align="middle">
               <Button
                 onClick={this.showAddVolumeLayerModal}
-                style={{
```
Comment (hotzenklotz) [inline frontend/javascripts/oxalis/view/left-border-tabs/layer_settings_tab.tsx:1063]: The button was cutting off the text. I guess another side-effect of the new font. I don't think a strict width is necessary here.

File: frontend/javascripts/oxalis/view/components/markdown_modal.tsx [hunk 3]
```diff
@@ -77,7 +78,14 @@ export function MarkdownModal({
...
             }}
-            isTextArea
+
+            // style={title == null ? style : undefined}
+            // onChange={this.handleChange}
+            // onFocus={this.handleFocus}
+            // onBlur={this.handleBlur}
+            // value={this.state.currentValue}
+            // onPressEnter={onPressEnter}
+            // onKeyDown={this.blurOnEscape}
```
Comment (philippotto) [inline frontend/javascripts/oxalis/view/components/markdown_modal.tsx:88]: why is this commented?

**Review by philippotto — COMMENTED** [review]
great clean up :+1: one thing is probably still to do.

File: frontend/javascripts/oxalis/view/components/markdown_modal.tsx (hunk 3 above)
Comment (hotzenklotz) [reply frontend/javascripts/oxalis/view/components/markdown_modal.tsx:88]: Ups. I need to remove this. Just wanted to compare the props supported by <InputComponent> and <TextArea> and double check that no callback was lost.

**Review by philippotto — APPROVED** [review]
:100:

**Pull request merged**